### PR TITLE
rewrite reader for flink-1.13

### PR DESCRIPTION
**Pull request by humengyu2012**

### CODE REVIEW

**Comment by xuanyu66** [comment]
Plz add description

File: flink/flink-1.13/src/main/java/io/tidb/bigdata/flink/connector/source/reader/TiDBSourceReader.java [hunk 1]
```diff
@@ -16,48 +16,192 @@
...
+  /**
+   * The availability future. This reader is available as soon as a split is assigned.
+   */
+  private CompletableFuture<Void> availability;
 
-public class TiDBSourceReader extends
-    SingleThreadMultiplexSourceReaderBase<RowData, RowData,
-        TiDBSourceSplit, TiDBSourceSplitState> {
+  private TiDBSourceSplit currentSplit;
+  private long offset;
```
Comment (xuanyu66) [inline flink/flink-1.13/src/main/java/io/tidb/bigdata/flink/connector/source/reader/TiDBSourceReader.java:64]: If there any overflow potential?

**Review by xuanyu66 — CHANGES_REQUESTED** [review]

File: flink/flink-1.13/src/test/java/io/tidb/bigdata/flink/tidb/FlinkTest.java [hunk 2]
```diff
@@ -430,4 +386,95 @@ public void testSnapshotRead() throws Exception {
...
+        String.format("SPLIT TABLE `%s` BETWEEN (0) AND (2000) REGIONS %s", tableName, 2),
+        String.format("INSERT INTO `%s` VALUES %s", tableName, values));
+    EnvironmentSettings settings = EnvironmentSettings.newInstance().useBlinkPlanner()
+        .inStreamingMode().build();
+    StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+    env.enableCheckpointing(100L);
+    StreamTableEnvironment tableEnvironment = StreamTableEnvironment.create(env, settings);
+    env.setParallelism(1);
+    properties.put("type", "tidb");
+    properties.put(TiDBOptions.SOURCE_FAILOVER.key(), "offset");
```
Comment (shiyuhang0) [inline flink/flink-1.13/src/test/java/io/tidb/bigdata/flink/tidb/FlinkTest.java:453]: Do we need test the default value

**Review by shiyuhang0 — APPROVED** [review]

File: flink/flink-1.13/src/test/java/io/tidb/bigdata/flink/tidb/FlinkTest.java (hunk 2 above)
Comment (xuanyu66) [reply flink/flink-1.13/src/test/java/io/tidb/bigdata/flink/tidb/FlinkTest.java:453]: See testReadAtLeastOnce(), the method above this

**Comment by humengyu2012** [comment]
see https://github.com/tidb-incubator/TiBigData/pull/168

**Pull request closed without merging**